app/services/cart_items_management_service.rb

```diff
@@ -0,0 +1,57 @@
+class CartItemsManagementService
+  attr_reader :cart, :product, :quantity
+
+  def initialize(cart, product, quantity)
+    @cart = cart
+    @product = product
+    @quantity = quantity
+  end
+
+  def add_to_cart
+    validate_quantity!
+
+    cart_product = CartProduct.find_or_initialize_by(cart: @cart, product: @product)
```

Comment:
you should use the attr_readers created

```suggestion
    cart_product = CartProduct.find_or_initialize_by(cart: cart, product: product)
```

Reply:
Updated.